### PR TITLE
Add checksum for py-setuptools-rust 1.2.0 and 1.4.1

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-setuptools-rust/package.py [hunk 1]
```diff
@@ -12,6 +12,8 @@ class PySetuptoolsRust(PythonPackage):
     homepage = "https://github.com/PyO3/setuptools-rust"
     pypi = "setuptools-rust/setuptools-rust-0.12.1.tar.gz"
 
+    version("1.4.1", sha256="18ff850831f58ee21d5783825c99fad632da21e47645e9427fd7dec048029e76")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-setuptools-rust/package.py:15]: Confirmed sha256.

File: var/spack/repos/builtin/packages/py-setuptools-rust/package.py [hunk 2]
```diff
@@ -12,6 +12,8 @@ class PySetuptoolsRust(PythonPackage):
...
+    version("1.2.0", sha256="a47d72d3723d03c7f66c283e301bbbf4260c2a0e7f02cda30ac2b8f65e42720b")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-setuptools-rust/package.py]: This sha256 is incorrect.

```suggestion
    version("1.2.0", sha256="0a4ada479e8c7e3d8bd7cb56e1a29acc2b2bb98c2325051b0cdcb57d7f056de8")
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please correct the checksum for 1.2.0

File: var/spack/repos/builtin/packages/py-setuptools-rust/package.py [hunk 3]
```diff
@@ -22,10 +24,15 @@ class PySetuptoolsRust(PythonPackage):
...
     depends_on("python@3.6:", when="@0.12:", type=("build", "run"))
+    depends_on("py-setuptools@62.4:", when="@1.4.0:", type=("build", "run"))
     depends_on("py-setuptools@46.1:", when="@0.11.6:", type=("build", "run"))
     depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-setuptools-scm+toml@6.3.2:", when="@1.2.0:", type="build")
     depends_on("py-setuptools-scm+toml@3.4.3:", when="@0.11:", type="build")
+    depends_on("py-semantic-version@2.8.2:2", when="@1.2.0:", type=("build", "run"))
     depends_on("py-semantic-version@2.6.0:", type=("build", "run"))
+    depends_on("py-typing-extensions@3.7.4.3:", when="@1.2.0:", type=("build", "run"))
     depends_on("py-toml@0.9.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-setuptools-rust/package.py]: Seems like toml is no longer required

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**